src/main/java/neqsim/thermo/phase/PhaseEos.java

```diff
@@ -214,41 +208,31 @@ public double molarVolume2(double pressure, double temperature, double A, double
       BonV = 1.0;
     }
     double BonVold = BonV;
-    double Btemp = 0;
-    double Dtemp = 0;
-    double h = 0;
-    double dh = 0;
-    double gvvv = 0;
-    double fvvv = 0;
-    double dhh = 0;
-    double d1 = 0;
-    double d2 = 0;
-    Btemp = getB();
-    Dtemp = getA();
+    double Btemp = getB();
+    double Dtemp = getA();
 
     setMolarVolume(1.0 / BonV * Btemp / numberOfMolesInPhase);
     int iterations = 0;
     int maxIterations = 1000;
-
     do {
       iterations++;
       BonVold = BonV;
-      h = BonV + Btemp * gV() + Btemp * Dtemp / (numberOfMolesInPhase * temperature) * fv()
+      double h = BonV + Btemp * gV() + Btemp * Dtemp / (numberOfMolesInPhase * temperature) * fv()
```

Comment:
will this increase calculation time (making doubles inside the while loop)?

Reply:
I guess the only reason to have it as before is speed (this loop is run a lot of times in all types of calculations) - I guess in any calculations a significant amount of time is spent on this method.

Reply:
I was hoping the compiler would optimize the code such that it did not matter. 

https://stackoverflow.com/questions/8803674/declaring-variables-inside-or-outside-of-a-loop